Adding the chat feature

DanielMy777 opened the pull request:
# Description
1) Adding url path to the chat for each individual connection.
2) Adding view to handle error and logic of chat access.
3) Adding the template of the chat.
4) Adding tests for the new feature.
5) In addition, for the images to show properly in the chat an image_url field was added to the user model in the admin page to allow change.

## Screenshots
![Capture](https://user-images.githubusercontent.com/79100490/119110875-4a58cf80-ba01-11eb-9f8c-f2d02867adbd.PNG)


## Manual Tests
Tried to manually force all errors and received positive feedback.
Tried to view the messages from both seeker and owner view.
Tried to add new messages seeing them update in the chat.

## Additional Information
NOTE: you still cant send messages, only view existing ones, the PR was heavy therefor will be split.
           you also cant access chats from the sidebar yet. both these improvements will take place in the next PR.


### Issues closed by this PR
fix #216 
fix #217 
### PR Dependencies
THIS PR WILL NEED A REBASE AFTER #219 IS MERGED

Micha546 commented on users/admin.py:
```diff
@@ -14,7 +14,7 @@ class UserAdminConfig(UserAdmin):
         ('Overview', {'fields': ('email', 'first_name', 'last_name',
                                  'birth_date', 'date_joined',
-                                 'is_active', 'hobbies')}),
+                                 'is_active', 'hobbies', 'image_url')}),
```
Nice!

Micha546 reviewed (approved):
🔥 👑

AmitAharoni reviewed (approved):
I'm standing and clapping my hands!
Well done!!

DanielMy777 committed:
Adding image field to admin user page

DanielMy777 committed:
Adding the chat feature

1) Adding url path to the chat for each individual connection.
2) Adding view to handle error and logic of chat access.
3) Adding the template of the chat.
